test/test.js

```diff
@@ -19083,6 +19083,29 @@
 
   /*--------------------------------------------------------------------------*/
 
+  QUnit.module('lodash.toSafeInteger');
+
+  (function() {
+    QUnit.test('should convert values to safe integers', function(assert) {
+      assert.expect(6);
+
+      assert.strictEqual(_.toSafeInteger(-5.6), -5);
+      assert.strictEqual(_.toSafeInteger('5.6'), 5);
+      assert.strictEqual(_.toSafeInteger(), 0);
+      assert.strictEqual(_.toSafeInteger(NaN), 0);
```

Comment:
> If we're following that spec, why do we want to deviate on infinity and
-infinity? For safeness?

The internal `ToInteger` doesn't enforce that the return value is an integer. It gets away with it because it's internal and not exposed. Our `_.toInteger` on the other hand is exposed and anything named `toThing` should always return some `thing`.